hoot-core/src/main/cpp/hoot/core/io/HootApiDb.h

```diff
@@ -278,6 +279,29 @@ class HootApiDb : public ApiDb
 
   virtual long getNextId(const ElementType& elementType);
 
+  /**
+   * Updates an existing job status; assumes the job record already exists
+   *
+   * @param jobId ID of an existing job
+   * @param status status text to update the job status field with
+   */
+  void updateJobStatus(const QString jobId, const QString status);
+
+  /**
+   * Determines whether a job status exists
+   *
+   * @param jobId the ID of the job status to determine the existence of
+   * @return true if it exists; false otherwise
+   */
+  bool jobStatusExists(const QString jobId);
+
+  /**
+   * Inserts a new job status
+   *
+   * @param jobId the ID of the newly inserted job status
+   */
+  void insertJobStatus(const QString jobId);
+
 protected:
```

Comment:
Oh, is that how its done?  That would have probably been easier than what I've done :-(  Let me take a look at the export script more closely and see how it works...then maybe try to conform to that method.  The part I'm unclear on is since its asynchronous, how the services know to package the output files in that fashion...unless that logic is in the script itself.